Assets/Tests/PlayModeTests/SessionTests.cs

```diff
@@ -0,0 +1,301 @@
+
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            CountlyRequestModel requestModel = Countly.Instance.Session._requestCountlyHelper._requestRepo.Dequeue();
+            string myUri = requestModel.RequestUrl;
+            NameValueCollection values = HttpUtility.ParseQueryString(myUri);
+
+
+            Assert.AreEqual("1", values.Get("begin_session"));
+            Assert.IsNotNull(values.Get("metrics"));
+        }
+
+        /// <summary>
+        /// It validates the of session service when autmatic session tracking is disable.
+        /// </summary>
+        [Test]
+        public void TestSessionService_WithDisableTracking()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+            };
+
+            configuration.DisableAutomaticSessionTracking();
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+        }
+
+        /// <summary>
+        /// It validates the functionality of 'BeginSessionAsync' when session consent is given after init and automatic session tracking is disable.
+        /// </summary>
+        [Test]
+        public void TestSessionBegin_ConsentGivenAfterInitWithDisableTracking()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+                RequiresConsent = true
+            };
+
+            configuration.DisableAutomaticSessionTracking();
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+           
+        }
+
+        /// <summary>
+        /// It validates the functionality of 'BeginSessionAsync' when session consent is given after init and session had begun before.
+        /// </summary>
+        [Test]
+        public void TestSessionBegin_ConsentGivenAfterSessionBegin()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+                RequiresConsent = true
+            };
+
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
```

Comment:
this looks wrong, why is this value "1"?

Reply:
It's not, there will an empty location request in RQ, "Location consent is not given".